client/dashboard/sites/backup-restore/form.tsx

```diff
@@ -0,0 +1,149 @@
+import { useMutation } from '@tanstack/react-query';
+import { Button, __experimentalHStack as HStack } from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { DataForm } from '@wordpress/dataviews';
+import { __ } from '@wordpress/i18n';
+import { rotateLeft } from '@wordpress/icons';
+import { store as noticesStore } from '@wordpress/notices';
+import { useState } from 'react';
+import { siteBackupRestoreInitiateMutation } from '../../app/queries/site-backup-restore';
+import { siteBackupRestoreRoute } from '../../app/router/sites';
+import Notice from '../../components/notice';
+import type { RestoreConfig } from '../../data/site-backup-restore';
+import type { Field } from '@wordpress/dataviews';
+
+const fields: Field< RestoreConfig >[] = [
+	{
+		id: 'themes',
+		label: __( 'WordPress themes' ),
+		Edit: 'checkbox',
+	},
+	{
+		id: 'plugins',
+		label: __( 'WordPress plugins' ),
+		Edit: 'checkbox',
+	},
+	{
+		id: 'roots',
+		label: __( 'WordPress root' ),
+		description: __( 'Includes wp-config php and any non WordPress files.' ),
+		Edit: 'checkbox',
+	},
+	{
+		id: 'contents',
+		label: __( 'WP-content directory' ),
+		description: __( 'Excludes themes, plugins, and uploads.' ),
+		Edit: 'checkbox',
+	},
+	{
+		id: 'sqls',
+		label: __( 'Site database' ),
+		description: __( 'Includes pages, and posts.' ),
+		Edit: 'checkbox',
+	},
+	{
+		id: 'uploads',
+		label: __( 'Media uploads' ),
+		description: __( 'You must also select Site database for restored media uploads to appear.' ),
+		Edit: 'checkbox',
+	},
+];
+
+function SiteBackupRestoreForm( {
+	siteId,
+	onRestoreInitiate,
+}: {
+	siteId: number;
+	onRestoreInitiate: ( restoreId: number ) => void;
+} ) {
+	const { rewindId } = siteBackupRestoreRoute.useParams();
+	const { mutate: restoreMutation, isPending: isRestoreMutationPending } = useMutation(
+		siteBackupRestoreInitiateMutation( siteId )
+	);
+	const { createErrorNotice } = useDispatch( noticesStore );
+
+	const [ formData, setFormData ] = useState< RestoreConfig >( {
+		themes: true,
+		plugins: true,
+		roots: true,
+		contents: true,
+		sqls: true,
+		uploads: true,
+	} );
+
+	const form = {
+		type: 'regular' as const,
+		fields: [ 'themes', 'plugins', 'roots', 'contents', 'sqls', 'uploads' ],
+	};
+
+	const handleFormChange = ( changes: Partial< RestoreConfig > ) => {
+		setFormData( ( prev ) => ( {
+			...prev,
+			...Object.fromEntries(
```

Comment:
Is there any reason we need to check whether the value is `undefined` or not 🤔 Instead of

```js
onChange={ ( edits: Partial<RestoreConfig> ) => {
	setFormData( ( data ) => ( { ...data, ...edits } ) );
}
```

Reply:
Not at all, just simplified it on b88683d4e2feed1f02f99d72b34a2f11f1763aaa